src/Storages/System/StorageSystemDetachedParts.cpp

```diff
@@ -230,6 +231,19 @@ class DetachedPartsSource : public ISource
                 size_t bytes_on_disk = parts_sizes.at(p_id - begin).load();
                 new_columns[res_index++]->insert(bytes_on_disk);
             }
+            if (columns_mask[src_index++])
+            {
+                Poco::Timestamp modification_time;
```

Comment:
Looks suspicious - initialize to zero.

Reply:
Actually the default ctor of the Timestamp uses "now", and that was the intention

And your change (`s/Poco::Timestamp modification_time;/Poco::Timestamp modification_time{};/`) did not change anything, there will still be "now" if `stat` fails.

Reply:
Thank you! I was only reading the diff, and the code was suspicious for me.